### PR TITLE
[fix](full compaction) Full compaction should hold meta lock when modifying tablet's meta data

### PR DESCRIPTION
**Pull request by bobhan1**
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

File: be/src/olap/full_compaction.cpp [hunk 1]
```diff
@@ -117,8 +117,11 @@ Status FullCompaction::modify_rowsets(const Merger::Statistics* stats) {
                 _full_compaction_update_delete_bitmap(_output_rowset, _output_rs_writer.get()));
     }
     std::vector<RowsetSharedPtr> output_rowsets(1, _output_rowset);
-    RETURN_IF_ERROR(_tablet->modify_rowsets(output_rowsets, _input_rowsets, true));
-    _tablet->save_meta();
+    {
+        std::lock_guard<std::shared_mutex> wrlock(_tablet->get_header_lock());
```
Comment (zhannngchen) [inline be/src/olap/full_compaction.cpp:121]: should hold `rowset_update_lock`

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by wm1581066 — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**